integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/utils/PlanUtils3.java

```diff
@@ -135,4 +143,32 @@ public static <D extends OpenLineage.Dataset> List<D> fromDataSourceV2Relation(
 
     return Collections.singletonList(datasetFactory.getDataset(di.get(), datasetFacetsBuilder));
   }
+
+  public static Optional<LogicalPlan> getLogicalPlanOf(
```

Comment:
Please consider adding comments here that explain in more details how SharedState and CacheManager interact with InMemoryRelation.

Reply:
Sure, good point!